receiver/MessageHandlingStrategy.py

```diff
@@ -0,0 +1,6 @@
+from typing import Protocol
+
+
+class MessageHandlingStrategy(Protocol):
+    async def on_message(self, data: dict) -> None:
+        pass
```

Comment:
Not sure if the `Protocol` class is necessary if you only have a single function. Maybe `ABC` is more fitting here?

Reply:
The difference really is negligible.
	However, ABC does enforce its interface which is desired
	in this case.
	Protocol is when inheritance is bad, but there is value in defining the required structure.
	In this case since structure is strictly defined & inheritance would make sense,
		you are right, ABC is the better choice.